pkg/api/microservices/microservices_test.go

```diff
@@ -0,0 +1,159 @@
+		"test-pair-1": testutil.GetModelBindingPair(),
+	}
+	// GIVEN empty model binding pairs, managed clusters and fake kubernetes clients.
+	var clients kubernetes.Interface = k8sfake.NewSimpleClientset()
+	clusters := []v1beta1.VerrazzanoManagedCluster{}
+
+	type args struct {
+		w http.ResponseWriter
+		r *http.Request
+	}
+	tests := []struct {
+		name    string
+		listers controller.Listers
+	}{
+		{
+			name:    "verifyReturnSingleMicroservice",
+			listers: testutilcontroller.NewControllerListers(&clients, clusters, &modelBindingPairs),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			vars := map[string]string{
+				"id": "HelidonApp1",
+			}
+			var myMicroservice Microservice
+			Init(tt.listers)
+			resp := httptest.NewRecorder()
+			req := httptest.NewRequest("GET", "/microservices", nil)
+			req = mux.SetURLVars(req, vars)
+			ReturnSingleMicroservice(resp, req)
+			assert.Equal(t, http.StatusOK, resp.Code, "expect the http return code to be http.StatusOk")
+			json.NewDecoder(resp.Body).Decode(&myMicroservice)
+			assert.Equal(t, "HelidonApp1", myMicroservice.ID, "expect the Microservice with ID = HelidonApp1 to be successfully returned")
+			assert.Equal(t, "HelidonApp1", myMicroservice.Name, "expect Microservice Name to be HelidonApp1")
+			assert.Equal(t, "cluster1", myMicroservice.Cluster, "expect Helidon App Cluster Name to be cluster1")
+			assert.Equal(t, "defaultOnMC1", myMicroservice.Namespace, "expect Helidon App Namespace to be defaultOnMC1")
+		})
+	}
+}
+
+// TestReturnSingleMicroserviceNotFound tests that a Microservice that doesn't exist is not found
+// GIVEN that Microservice with ID=HelidonApp100 doesn't exist
+// WHEN I call ReturnSingleMicroservice
+// THEN no Microservice should be returned
+// AND that the MicroService returned should be empty
+func TestReturnSingleMicroserviceNotFound(t *testing.T) {
+	var modelBindingPairs = map[string]*types.ModelBindingPair{
+		"test-pair-1": testutil.GetModelBindingPair(),
+	}
+	// GIVEN empty model binding pairs, managed clusters and fake kubernetes clients.
+	var clients kubernetes.Interface = k8sfake.NewSimpleClientset()
+	clusters := []v1beta1.VerrazzanoManagedCluster{}
+
+	type args struct {
+		w http.ResponseWriter
+		r *http.Request
+	}
+	tests := []struct {
+		name    string
+		listers controller.Listers
+	}{
+		{
+			name:    "verifyReturnSingleMicroserviceNotFound",
+			listers: testutilcontroller.NewControllerListers(&clients, clusters, &modelBindingPairs),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			vars := map[string]string{
+				"id": "HelidonApp100",
+			}
+			var myMicroservice Microservice
+			Init(tt.listers)
+			resp := httptest.NewRecorder()
+			req := httptest.NewRequest("GET", "/microservices", nil)
+			req = mux.SetURLVars(req, vars)
+			ReturnSingleMicroservice(resp, req)
+			assert.Equal(t, http.StatusOK, resp.Code, "expect the http return code to be http.StatusOk")
```

Comment:
the API _should_ be fixed to return 404 here. I had to fix a couple of those in another API when adding the tests.

Reply:
Are we supposed to be fixing the API?

Reply:
did i miss the memo

Reply:
As we discussed yesterday, my preference is to fix the API while we are here. I know of a few cases where we did do this